src/graph/client/graphClient.ts

```diff
@@ -273,16 +286,176 @@ export class GraphClient {
     }
 
     this.setStateResults(true);
-    this.displayGraph(results.countUniqueVertices, results.limitedVertices, results.countUniqueEdges, results.limitedEdges);
+    this._graphView.display(results.countUniqueVertices, results.limitedVertices, results.countUniqueEdges, results.limitedEdges);
   }
 
   private splitVerticesAndEdges(nodes: any[]): [GraphVertex[], GraphEdge[]] {
     let vertices = nodes.filter(n => n.type === "vertex");
     let edges = nodes.filter(n => n.type === "edge");
     return [vertices, edges];
   }
+}
+
+class GraphView {
+  private _countUniqueVertices: number;
+  private _vertices: GraphVertex[];
+  private _countUniqueEdges: number;
+  private _edges: GraphEdge[];
+  private _force: any;
+
+  public display(countUniqueVertices: number, vertices: GraphVertex[], countUniqueEdges: number, edges: GraphEdge[]) {
+    this._countUniqueVertices = countUniqueVertices;
+    this._vertices = vertices;
+    this._countUniqueEdges = countUniqueEdges;
+    this._edges = edges;
+
+    this.clear();
+
+    // Set up nodes and links for the force simulation
```

Comment:
All I did here was move the code and also remove a try/catch (thus all the detected whitespace changes), which was moved to the generic server message handler.